test/integration/controlplane/transformation/kms_transformation_test.go

```diff
@@ -515,35 +516,81 @@ resources:
 	previousConfigDir := test.configDir
 	test.shutdownAPIServer()
 	restarted = true
-	test, err = newTransformTest(t, "", true, previousConfigDir, false)
+	test, err = newTransformTest(t, "", true, previousConfigDir)
 	if err != nil {
 		t.Fatalf("failed to start KUBE API Server with encryptionConfig\n %s, error: %v", encryptionConfig, err)
 	}
-	test.cleanUp()
+	defer test.cleanUp()
+
+	// confirm that reading cluster wide secrets still works after restart
+	_, err = test.restClient.CoreV1().Secrets("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("failed to list secrets, err: %v", err)
+	}
+
+	// make sure cluster wide configmaps read still works
+	_, err = test.restClient.CoreV1().ConfigMaps("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("failed to list configmaps, err: %v", err)
+	}
 }
 
 func TestEncryptionConfigHotReloadFileWatch(t *testing.T) {
 	testCases := []struct {
-		fileUpdateMethod string
-		sleep            time.Duration
+		sleep      time.Duration
+		name       string
+		updateFile func(filePath, fileContent string) error
 	}{
 		{
-			fileUpdateMethod: "truncate",
-			sleep:            20 * time.Second, // significantly longer than KMSCloseGracePeriod
-		},
-		{
-			fileUpdateMethod: "deleteAndCreate",
+			name: "truncate file",
+			updateFile: func(filePath string, fileContent string) error {
+				// os.WriteFile truncates the file before writing
+				return os.WriteFile(filePath, []byte(fileContent), 0644)
+			},
+			// significantly longer than KMSCloseGracePeriod
+			sleep: 20 * time.Second,
 		},
 		{
-			fileUpdateMethod: "move",
+			name: "delete and create file",
+			updateFile: func(filePath, fileContent string) error {
+				// os.Remove deletes the file before creating a new one
+				if err := os.Remove(filePath); err != nil {
+					return fmt.Errorf("failed to remove encryption config, err: %w", err)
+				}
+
+				file, err := os.Create(filePath)
+				if err != nil {
+					return fmt.Errorf("failed to create encryption config, err: %w", err)
+				}
+				defer file.Close()
+
+				if _, err := file.Write([]byte(fileContent)); err != nil {
+					return fmt.Errorf("failed to write encryption config, err: %w", err)
+				}
+
+				return nil
+			},
 		},
 		{
-			fileUpdateMethod: "symLink",
+			name: "move file",
+			updateFile: func(filePath, fileContent string) error {
+				// write new config to a temp file
+				if err := os.WriteFile(filePath+".tmp", []byte(fileContent), 0644); err != nil {
```

Comment:
make `filePath + ".tmp"` a variable and reuse here and https://github.com/kubernetes/kubernetes/pull/113896/files#diff-757c6bfc7bd92dcf0f33365f70feea4a5ceafad120668a60ff2934941ece79efR583